cinema/views.py

```diff
@@ -19,27 +20,48 @@
     OrderSerializer,
     OrderListSerializer,
 )
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
 
 
 class GenreViewSet(viewsets.ModelViewSet):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
+
 
 class ActorViewSet(viewsets.ModelViewSet):
     queryset = Actor.objects.all()
     serializer_class = ActorSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
+
 
 class CinemaHallViewSet(viewsets.ModelViewSet):
     queryset = CinemaHall.objects.all()
     serializer_class = CinemaHallSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
+
 
 class MovieViewSet(viewsets.ModelViewSet):
     queryset = Movie.objects.prefetch_related("genres", "actors")
     serializer_class = MovieSerializer
 
+    def get_permissions(self):
+        if self.action in ("list", "create", "retrieve"):
+            return (IsAdminOrIfAuthenticatedReadOnly(),)
+        return (IsAuthenticated(),)
```

Comment:
The `get_permissions` method in `MovieViewSet` includes 'retrieve' in the actions that allow read-only access for authenticated users. Ensure this is intended, as it might expose detailed movie information to all authenticated users.